src/core/main.lua

```diff
@@ -172,17 +172,19 @@ end
 
 -- Cleanup after Snabb process.
 function shutdown (pid)
+   -- simple pcall helper to print error and continue
+   local function safely (f)
+      local ok, err = pcall(f)
+      if not ok then print(err) end
+   end
+   -- Run cleanup hooks
+   safely(function () require("apps.interlink.receiver").shutdown(pid) end)
```

Comment:
Honestly, I don’t see the difference between core calling into lib/ vs apps/, its both outside core. As it stands, lib/interlink and apps/interlink form a single whole, the reason the shutdown hooks are in apps/interlink is because that is where SHM paths are chosen.

Reply:
Alternately we could move lib.interlink into core, and make it choose the SHM paths.  WDYT?